src/ClientData/ScopeIdProvider.cpp

```diff
@@ -48,10 +61,17 @@ namespace orbit_client_data {
 
     uint64_t id = next_id_;
     name_to_id_[key] = id;
+    scope_id_to_name_[id] = timer_info.api_scope_name();
     next_id_++;
     return id;
   }
   return orbit_client_data::kInvalidScopeId;
 }
 
+const std::string& NameEqualityScopeIdProvider::GetScopeName(uint64_t scope_id) const {
+  const auto it = scope_id_to_name_.find(scope_id);
+  ORBIT_CHECK(it != scope_id_to_name_.end());
+  return it->second;
```

Comment:
personal (not strong) preference
```suggestion
  ORBIT_CHECK(scope_id_to_name_.contains(scope_id));
  return scope_id_to_name_.at(scope_id);
```

Reply:
That means performing the lookup twice.